### ISSUE TITLE
Incorrect variance in normalization layer of EfficientNet

### ISSUE THREAD

**Issue by kaczmarj** [tensorflow/tensorflow#49930]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04.5 LTS (Google Colab)
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: n/a
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): v2.5.0-0-ga4dfb8d1a71 2.5.0
- Python version: 3.7.10
- Bazel version (if compiling from source): n/a
- GCC/Compiler version (if compiling from source): n/a
- CUDA/cuDNN version: n/a
- GPU model and memory: n/a

**Describe the current behavior**

EfficientNet includes [a normalization layer within its model definition](https://github.com/tensorflow/tensorflow/blob/a4dfb8d1a71385bd6d122e4f27f86dcebb96712d/tensorflow/python/keras/applications/efficientnet.py#L321), but it seems like the variance is incorrect. The variance is `[0.229, 0.224, 0.225]`, but [those values are the standard deviations of the ImageNet dataset](https://github.com/tensorflow/tensorflow/blob/a4dfb8d1a71385bd6d122e4f27f86dcebb96712d/tensorflow/python/keras/applications/imagenet_utils.py#L196). When the normalization layer is called on new inputs, the inputs are normalized using the mean (which looks correct) and the square root of the variance. So in the current EfficientNet implementation, inputs are normalized using the square root of the standard deviation of ImageNet.

**Describe the expected behavior**

I expect inputs to EfficientNet to be normalized according to the standard deviation of the ImageNet dataset.

**[Contributing](https://www.tensorflow.org/community/contribute)** - Do you
want to contribute a PR? (yes/no): no, because I don't know where I would make this change. The change would have to be within the saved models.

**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.

```python
import numpy as np
import tensorflow as tf

model = tf.keras.applications.EfficientNetB0(weights="imagenet")
norm_layer = model.layers[2]
assert "normalization" in norm_layer.name
print(norm_layer.mean.numpy())  # [0.485 0.456 0.406]
print(norm_layer.variance.numpy())  # [0.229 0.224 0.225]
# Generate sample inputs.
tf.random.set_seed(42)
x = tf.random.uniform((1, 224, 224, 3), 0, 255, dtype="int32", seed=42)


def get_reference(inputs):
    """Get the reference normalized outputs."""
    x = np.asarray(inputs).astype("float32")
    x /= 255.0
    x[..., 0] -= 0.485
    x[..., 1] -= 0.456
    x[..., 2] -= 0.406
    x[..., 0] /= 0.229
    x[..., 1] /= 0.224
    x[..., 2] /= 0.225
    return x


def get_current_tf_efficientnet_norm_output(inputs):
    """Get the normalized outputs from the current implementation."""
    x = np.asarray(inputs).astype("float32")
    x /= 255.0
    x[..., 0] -= 0.485
    x[..., 1] -= 0.456
    x[..., 2] -= 0.406
    x[..., 0] /= np.sqrt(0.229)
    x[..., 1] /= np.sqrt(0.224)
    x[..., 2] /= np.sqrt(0.225)
    return x


model_normalizer = tf.keras.Model(model.input, norm_layer.output)
# Below is True (they are the same)
np.allclose(get_current_tf_efficientnet_norm_output(x), model_normalizer(x), atol=1e-07)
# Below is False (they are different)
np.allclose(get_reference(x), model_normalizer(x), atol=1e-07)
```

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

The [Normalization layer normalizes using the square root of the variance](https://github.com/tensorflow/tensorflow/blob/a4dfb8d1a71385bd6d122e4f27f86dcebb96712d/tensorflow/python/keras/layers/preprocessing/normalization.py#L242-L243) (which equal to the standard deviation)

**Comment by rmothukuru** [comment]
Could replicate the issue with **`Tensorflow Version 2.5`**. Please find [the Gist](https://colab.research.google.com/gist/rmothukuru/9adcbfed857b3fd011c67251d11e2147/gh_49930.ipynb). Thanks!

**Comment by qlzh727** [comment]
Thanks for reporting the issue. When I look at the original implementation from the paper in https://github.com/tensorflow/tpu/blob/master/models/official/efficientnet/efficientnet_model.py#L505, I didn't find any normalization layer for the inputs. (The rescale in expected in keras.application model to mimic https://github.com/tensorflow/tensorflow/blob/a4dfb8d1a71385bd6d122e4f27f86dcebb96712d/tensorflow/python/keras/applications/imagenet_utils.py#L190).

@fchollet, since the original implementation is added by you, could u provide more context about why the normalization layer is added?

**Comment by fanstvn** [comment]
My experiment using tf.keras.applications.EfficientNetB0 with TensorFlow 2.4.2 on ILSVRC2012 validation set only achieve ~74.6% of accuracy.
After fixing this problem by:
`    mean, std, a = model.layers[2].get_weights()
    model.layers[2].set_weights([mean, std**2, a])`
the result increased to 77.2%, as in the paper.

**Comment by fanstvn** [comment]
The Keras implemention of EfficientNet is a re-implemention of [reference code](https://github.com/tensorflow/tpu/tree/master/models/official/efficientnet). Normalization function and args can be found at [here](https://github.com/tensorflow/tpu/blob/abdfd6726cec215e9c2700d6f494f0a08077a2b8/models/official/efficientnet/main.py#L369) and [here](https://github.com/tensorflow/tpu/blob/b24729de804fdb751b06467d3dce0637fa652060/models/official/efficientnet/efficientnet_builder.py#L31).
It normalizes the data by `x = (x - mean * 255) / (std * 255)` and can be split into 2 steps: `x = x / 255` for rescale and `x = (x - mean) / std` for normalization.
The normalization layer in tf.keras normalizes the data using mean and variance, and applys `x = (x - mean) / sqrt(variance)` to inputs, but it is standard deviation in current pre-trained weights of tf.keras.applications.efficientnet, so it applys `x = (x - mean) / sqrt(std)` to inputs, which is incorrect.

**Comment by jvishnuvardhan** [comment]
@fanstvn If there is any actionable PRs, please feel free to open them in [keras-team/keras](https://github.com/keras-team/keras/issues) repository. 

Please note that Keras development moved to keras-team/keras repository to focus entirely on only keras. Thanks!

**Comment by kaczmarj** [comment]
@jvishnuvardhan - ok -- the rescaling weights of the trained model will have to be updated. can you transfer this pr to the keras repository? this pr was made while keras was still being developed in this repository.

**Comment by qlzh727** [comment]
I will take a closer look and fix this issue. Sorry for the long wait.

**Comment by qlzh727** [comment]
I was able to verify the correctness of fix by following code snippet.

```
import tensorflow_datasets as tfds
import tensorflow as tf

model = tf.keras.applications.EfficientNetB0(weights="imagenet")

BATCH = 128

def resize_and_center_crop(image: tf.Tensor, size: int) -> tf.Tensor:
  h, w = tf.shape(image)[0], tf.shape(image)[1]

  # Figure out the necessary h/w.
  ratio = (tf.cast(size, tf.float32) / tf.cast(tf.minimum(h, w), tf.float32))
  h = tf.cast(tf.round(tf.cast(h, tf.float32) * ratio), tf.int32)
  w = tf.cast(tf.round(tf.cast(w, tf.float32) * ratio), tf.int32)
  image = tf.image.resize(image, [h, w])
  top = (h - size) // 2
  left = (w - size) // 2
  image = tf.image.crop_to_bounding_box(image, top, left, size, size)
  return image

def eval_preprocess(features):
  image = features["image"]
  label = features["label"]
  assert image.dtype == tf.uint8
  image = tf.cast(image, tf.float32)
  # image = tf.keras.applications.resnet50.preprocess_input(image)
  image = resize_and_center_crop(image, 224)
  label = tf.one_hot(label, 1000)
  return image, label

dataset_builder = tfds.builder("imagenet2012")
eval_ds = dataset_builder.as_dataset(split='validation')
eval_ds = eval_ds.map(eval_preprocess, num_parallel_calls=tf.data.AUTOTUNE).batch(BATCH)

norm_layer = model.layers[2]
assert "normalization" in norm_layer.name
print(norm_layer.mean.numpy())  # [0.485 0.456 0.406]
print(norm_layer.variance.numpy())  # [0.229 0.224 0.225]
existing_variance = norm_layer.variance.numpy()

correct_variance = tf.math.square(existing_variance)
norm_layer.variance = tf.convert_to_tensor(correct_variance)

model.compile(metrics=['categorical_accuracy'])
model.evaluate(eval_ds, verbose=1)

# validation acc is 77.0
```

**Comment by qlzh727** [comment]
This should be fixed now in keras.
